src/python/pants/build_graph/resources.py

```diff
@@ -20,6 +20,10 @@ class Resources(Target):
   :API: public
   """
 
+  @classmethod
```

Comment:
Hm... why did you end up needing a macro for this one? Doesn't look like it's necessary.